test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -3549,6 +3549,28 @@ def forward(self, mat1, mat2):
         mat2 = torch.randn(3, 3)
         self.run_test(M(), input=(mat1, mat2))
 
+    def test_where_with_bool_tensor(self):
+        class M(torch.nn.Module):
+            def forward(self, mat1, mat2):
+                out = torch.where(mat1 > 0, mat1, mat2)
```

Comment:
Just curious, output of the Greater op should already be a boolean. Does this also need cast?

Reply:
No, I added this test point because I did not see any unit test for the existing support for `where` (with bool input). This is strictly not needed for this PR but I added it just to improve coverage.